### PR TITLE
add expo plugin

### PR DESCRIPTION
**Pull request by mvanroon**
This pull requests enables the installation of this library in a custom [Expo development build](https://docs.expo.dev/develop/development-builds/introduction/).

This way developers can use this package without having to eject their Expo app.

### CODE REVIEW

File: plugin/src/withAddCalendarEvent.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    contactsPermission?: string
+}
+
+const withAddCalendarEvent: ConfigPlugin<void | ConfigPluginProps> = (config, props) => {
+    config = withInfoPlist(config, (config) => {
+        if (!config.modResults.NSCalendarsUsageDescription) {
+            config.modResults.NSCalendarsUsageDescription = props?.calendarPermission ?? DEFAULT_CALENDAR_PERMISSION;
+        }
+
+        if (!config.modResults.NSCalendarsUsageDescription) {
```
Comment (vonovak) [inline plugin/src/withAddCalendarEvent.ts:17]:
```suggestion
        if (!config.modResults.NSContactsUsageDescription) {
```

**Review by vonovak — CHANGES_REQUESTED** [review]
unexpected (not a very active project), but thank you! 😄

File: plugin/src/withAddCalendarEvent.ts (hunk 1 above)
Comment (mvanroon) [reply plugin/src/withAddCalendarEvent.ts]: woops - thanks!

**Comment by mvanroon** [comment]
I've tested our fork in our app both Android and iOS are working as expected. Merge, release and publish when you're ready!

File: .gitignore [hunk 2]
```diff
@@ -12,7 +12,6 @@ yarn-error.log
 
 # Xcode
 #
-build/
```
Comment (vonovak) [inline .gitignore:15]: why this change? the build artifacts should not be in git (the config plugin build included)

File: plugin/build/withAddCalendarEvent.d.ts
[the remark this answers is not in the document]
Comment (mvanroon) [reply plugin/build/withAddCalendarEvent.d.ts:1]: I can remove it, but make sure you build the plugin before publishing to npm

[the remark this answers is not in the document]
Comment (mvanroon) [reply plugin/build/withAddCalendarEvent.d.ts:1]: Also: most expo plugins also check in the build folder:

https://github.com/rnmapbox/maps/tree/main/plugin/build
https://github.com/expo/sentry-expo/tree/main/plugin/build
https://github.com/thebergamo/react-native-fbsdk-next/tree/master/plugin/build

**Comment by vonovak** [comment]
thank you!

**Comment by mvanroon** [comment]
@vonovak Thanks for merging and publishing. Could you also tag master?

**Comment by vonovak** [comment]
@mvanroon published another with a clean commit and tag

**Comment by mvanroon** [comment]
@vonovak seems like you published to npm without building the plugin first. Can you run `yarn build:plugin` and re-publish?

**Comment by vonovak** [comment]
done :)

**Comment by vonovak** [comment]
the plugin files would not be shipped even when included in the repo, they were ignored by npm :)

**Comment by mvanroon** [comment]
Right! Thanks for the quick fix and publish!

**Pull request merged**